RecoLocalCalo/HcalRecAlgos/src/MahiFit.cc

```diff
@@ -546,16 +544,18 @@ void MahiFit::phase1Debug(const HBHEChannelInfo& channelData, MahiDebugInfo& mdi
         mdi.itPulse[iTS] = nnlsWork_.pulseMat.col(iBX).coeff(iTS);
       }
     } else if (nnlsWork_.bxs.coeff(iBX) == pedestalBX_) {
+      mdi.pedEnergy = 0;
       mdi.pedEnergy = nnlsWork_.ampVec.coeff(iBX);
-    } else if (nnlsWork_.bxs.coeff(iBX) == -1) {
-      mdi.pEnergy = nnlsWork_.ampVec.coeff(iBX);
+    } else if (nnlsWork_.bxs.coeff(iBX) < 8) {
+      int ootIndex = nnlsWork_.bxs.coeff(iBX);
+      if (ootIndex > 0)
+        ootIndex += 2;
+      else
+        ootIndex += 3;
+      mdi.ootEnergy[ootIndex] = 0;
+      mdi.ootEnergy[ootIndex] = nnlsWork_.ampVec.coeff(iBX);
```

Comment:
similarly here, what is the purpose of L555?

Reply:
L555 is removed.

Reply:
L555 is removed.